### DISCUSSION TITLE
Building with Hugo

### DISCUSSION THREAD

**Discussion by vincentmaule** [McShelby/hugo-theme-relearn#794] [General]
Hello All,

Looking to hopefully learn a little more & this maybe an only @McShelby question.
Recently finished up with this [issue](https://github.com/McShelby/hugo-theme-relearn/issues/792).

Based on what I've experienced, it seems I am experiencing these same permissions [results](https://github.com/McShelby/hugo-theme-relearn/issues/734#issuecomment-1962897617) that [ScrumpyJack](https://github.com/ScrumpyJack) shared.

This led to a few questions:
 - What would be the cause for these permissions to have this issue? 
 - I could change it myself with just changing the permissions but for the sake of the next person that comes along, can we change the permissions in the repo themselves? 
 - Is there a reason these permissions are set that I'm unaware of?

*Note: I closed the issue as I did get Hugo to build said site & thought it would be better to take conversation here.*

**Comment by McShelby** [comment]
I haven't understood what trickery Hugo does when it creates the page. So `--noChmod` can solve this issue but I don't know the internals.

Basically it comes down to the fact, that the theme has files in it's static folder that get copied over to ` public` by Hugo in an early phase of the build. During the build, the theme tries to store modified versions of those files in the same location in `public`. That's when this issue can arise.

I have an idea how to fix it, but as it only happens to few users yet, I wasn't in a hurry.

**Reply by dahlsailrunner** [reply]
i'm having the same issue and still haven't been able to find a solution yet (tried a few of the things here).  Interested  in any suggestions or solutions you might have.

Happy to share details but it's failing trying to copy the custom css assets to `public`.

version 0.122.0 of hugo works fine.

**Reply by McShelby** [reply]
What did you try, what error message did you get? On what host are you building?

**Reply by dahlsailrunner** [reply]
Tried this:

```yaml
[params]
  # Change default color scheme with a variant one.
  themeVariantModifier =''
  mermaidZoom = true 
  disableInlineCopyToClipBoard = true 
  editURL = "https://github.com/MyCompany/arch-docs/blob/main/content/"  
  alwaysOpen = false
  collapsibleMenu = true
  disableLandingPageButton = true 

[[params.themeVariant]]
  identifier = 'custom-light'

[[params.themeVariant]]
  identifier = 'custom-dark'
```
Also tried using just `themeVariant` array in `params` as well (that was the original state of the `config.toml` file).  Line would have been:

```yaml
[params]
    themeVariant = ['custom-light', 'custom-dark']
```

Error was this:
```
ERROR Failed to publish Resource: open C:\Users\me\source\repos\arch-docs\public\css\theme-relearn-light.css: Access is denied.
ERROR Failed to publish Resource: open C:\Users\me\source\repos\arch-docs\public\css\theme-relearn-dark.css: Access is denied.
```

**Reply by dahlsailrunner** [reply]
Missed the "what host" question.  I'm building on Windows.  It sounds from a teammate of mine that it works on Mac with 0.124.1 although i think it requires `sudo`.

**Reply by McShelby** [reply]
Did you try `themeVariantModifier` as mentioned in your cited issues?

**Reply by dahlsailrunner** [reply]
I think so -- that's the first code snippet above.  Maybe I missed something though?

**Reply by McShelby** [reply]
You need to give it a value like `themeVariantModifier=".gen"`

**Reply by dahlsailrunner** [reply]
Yeah I just realized that after reading the comments in the doc (adding here for additional visibility / clarity):

```
# The theme modifies the variant stylesheets during build and tries to store
# them with the same file name. In certain installations it was observed, that
# Hugo could not overwrite these files due to permission issues. If you
# experience this, you can first try to set the `--noChmod` option for Hugo.
# If this still doesn't help you can set this option to eg. ".gen". This will
# be used as a suffix for these generated files, causing them to be newly
# created instead of overwriting the existing ones.
```
Once I did that (your suggestion of using a `.gen` value) it worked.  Thanks a lot and hope this helps others!  :)
